counter/src/Pages/CounterPage/containers/CounterContainer.js

```diff
@@ -0,0 +1,74 @@
+import React, {Component} from 'react';
+import CounterView from "../components/CounterView";
+
+//import CounterLayout from "../components/CounterLayout/CounterLayout";
+
+class CounterContainer extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            // parityType: "even",
+            countValue:0,
+        } /* автоматически завели наше состояние; конструктор класса, который устанавливает начальное состояние в this.state*/
+    }
+    /*все методы, которые будут вызываться на пользовательские действия: клик, изменения инпута, должны начинаться со слова handle */
+
+    handleIncrement = () => {
+        this.setState( (state) => {
+            const incrementedValue = state.countValue + 1;
+
+            return {
+                countValue: incrementedValue
+            }
+        })
+
+    }
+    handleDecrement = () => {
+        this.setState( (state) => {
+
+            const decrementedValue = Math.max(state.countValue - 1, 0);
+            return {
+                countValue: decrementedValue
+            }
+        })
+
```

Comment:
Issue is that your setState here will be called every time. Try to write logic in this way: Run setState methode only if <...your statement...>